libraries/AP_HAL_Linux/HAL_Linux_Class.cpp

```diff
@@ -265,6 +279,22 @@ void _usage(void)
 
 void HAL_Linux::run(int argc, char* const argv[], Callbacks* callbacks) const
 {
+#if CONFIG_HAL_BOARD_SUBTYPE == HAL_BOARD_SUBTYPE_LINUX_IMX
+    //some times ardupilot bootup,but cmd->"modprobe cdc_acm" create file node->"/dev/ttyACM0" Not yet!
+    //just dead loop until file node created.
+    while (1)
+    {
+        int uart = mcu.init("/dev/ttyACM0");
+        if (uart >= 0)
+            break;
+        printf("/dev/ttyACM0 open error,waiting...\n");
+        usleep(1000000);
+    }
+    //set imx6 board default path, The following code will change the path if user set cmd param
+    uartBDriver.set_device_path("/dev/ttymxc2");
+    uartCDriver.set_device_path("/dev/ttymxc1");
```

Comment:
these should be passed as command line argument.

Reply:
ok,i will remove it